### ISSUE TITLE
Handle is already closing error

### ISSUE THREAD

**Issue by danielleontiev** [lewis6991/gitsigns.nvim#638]
### Description

I was working in the repository and suddenly error starts to appear:

![image](https://user-images.githubusercontent.com/24596613/193246037-fb15842d-63a4-476b-80e3-6e525517e0a2.png)

Pressing ENTER supresses the error but it appears in ~15 seconds again. I do not know how to reproduce the issue, though

### Neovim version

NVIM v0.7.2 Build type: Release LuaJIT 2.1.0-beta3 Compiled by brew@Monterey  Features: +acl +iconv +tui See ":help feature-compile"     system vimrc file: "$VIM/sysinit.vim"   fall-back for $VIM: "/usr/local/Cellar/neovim/0.7.2_1/share/nvim"  Run :checkhealth for more info

### Operating system and version

Mac OS Moneterey 12.6

### Expected behavior

Error should not appeat

### Actual behavior

Error keeps appearing

### Minimal config

```Lua
vim.o.packpath = '/tmp/nvim/site'

local plugins = {
  gitsigns = 'https://github.com/lewis6991/gitsigns.nvim',
  -- ADD OTHER PLUGINS _NECESSARY_ TO REPRODUCE THE ISSUE
}

for name, url in pairs(plugins) do
  local install_path = '/tmp/nvim/site/pack/test/start/'..name
  if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system { 'git', 'clone', '--depth=1', url, install_path }
  end
end

require('gitsigns').setup{
  debug_mode = true, -- You must add this to enable debug messages
  -- ADD GITSIGNS CONFIG THAT IS _NECESSARY_ FOR REPRODUCING THE ISSUE
}

-- ADD INIT.LUA SETTINGS THAT IS _NECESSARY_ FOR REPRODUCING THE ISSUE
```

### Steps to reproduce

1. `nvim --clean -u minimal.lua`
2. ...

**Comment by lewis6991** [comment]
Ok thanks. This should be enough info for me to look into.

**Comment by danielleontiev** [comment]
Thank you!

**Comment by datanoise** [comment]
After this commit I am getting error: too many files open.

**Comment by lewis6991** [comment]
Do you get this error before the commit?

**Comment by datanoise** [comment]
No. Reverting this commit fixes the issue. An easy test is to run vim with and without this commit and see how many open files are opened by nvim process.

**Comment by lewis6991** [comment]
Does https://github.com/lewis6991/gitsigns.nvim/commit/754177a24803eb31684e79645733b3192fdcdd5d fix the issue?
